src/power_grid_model/_core/power_grid_model.py

```diff
@@ -120,6 +120,26 @@ def __deepcopy__(self, memo: dict[int, Any]) -> "PowerGridModel":
 
         return new_model
 
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the current model.
+        This includes the total number of components and the number of components per component type of the model.
+
+        Returns:
+            String representation of the model
+        """
+        try:
+            component_count = self.all_component_count
+        except TypeError:
+            component_count = {}
+
+        message = f"{self.__class__.__name__} ({sum(component_count.values())} components)\n"
```

Comment:
did not add a test case for this, but manually verified that this also works for derived classes :+1: